components/script/dom/window.rs

```diff
@@ -1803,18 +1786,14 @@ impl Window {
         ScriptThread::handle_tick_all_animations_for_testing(pipeline_id);
     }
 
-    /// Reflows the page unconditionally if possible and not suppressed. This
-    /// method will wait for the layout thread to complete (but see the `TODO`
-    /// below). If there is no window size yet, the page is presumed invisible
-    /// and no reflow is performed. If reflow is suppressed, no reflow will be
-    /// performed for ForDisplay goals.
-    ///
-    /// TODO(pcwalton): Only wait for style recalc, since we have
```

Comment:
Just for my understanding, is this TODO referring to an optimization that was possible with separate layout threads, but no longer possible after this change?

Reply:
Yes, unfortunately this was the unmet promise of the layout thread, but it was never implemented as far as I could tell. The thing is that the specifications lay down some pretty strict ordering requirements regarding the Document event loop and layout and furthermore even among different Documents that share the same origin and iframe tree structure. This is why all iframe tests are incredibly flaky in Servo.